src/ArtisanApartmentCommands.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace BuildEmpire\Apartment;
+
+use BuildEmpire\Apartment\Exceptions\SchemaAlreadyExistsException;
+use BuildEmpire\Apartment\Exceptions\SchemaCannotBePublicException;
+use BuildEmpire\Apartment\Exceptions\SchemaDoesntExistException;
+use BuildEmpire\Apartment\Exceptions\SchemaNameNotValidException;
+use BuildEmpire\Apartment\Schema;
+use Illuminate\Support\Facades\Schema as LumenSchema;
+use Carbon\Carbon;
+use BuildEmpire\Apartment\Helpers\ApartmentHelpers;
+
+class ArtisanApartmentCommands
+{
+    const ARTISAN = 1;
+    const WEB = 2;
+    const PUBLIC_SCHEMA = 'public';
+
+    protected $schema = false;
+
+    /**
+     * ArtisanApartmentCommands constructor.
+     * @param \BuildEmpire\Apartment\Schema $schema
+     */
+    public function __construct(Schema $schema)
+    {
+        $this->schema = $schema;
+    }
+
+    /**
+     * Make an apartment's schema and run any existing apartment migrations.
+     *
+     * @param $schemaName
+     * @throws SchemaAlreadyExistsException
+     * @throws SchemaCannotBePublicException
+     * @throws SchemaNameNotValidException
+     */
+    public function makeSchema($schemaName)
+    {
+
+        if (!ApartmentHelpers::isSchemaNameValid($schemaName)) {
+            throw new SchemaNameNotValidException('The apartment ' . $schemaName . ' is not valid. It must be all lowercase and only contain letters, numbers, or underscores.');
+        }
+
+        if ($this->schema->doesSchemaExist($schemaName)) {
+            throw new SchemaAlreadyExistsException('The apartment ' . $schemaName . ' already exists.');
+        }
+
+        if ($this->isSchemaPublic($schemaName)) {
+            throw new SchemaCannotBePublicException('The apartment name cannot be ' . self::PUBLIC_SCHEMA);
+        }
+
+        app('db')->transaction(function () use ($schemaName) {
+
+            $this->createSchema($schemaName);
+
+            $this->createSchemaMigrationTable($schemaName);
+
+            $this->createSchemaMetadata($schemaName);
+
+            $this->updateSchemaMetadata($schemaName, self::ARTISAN, Carbon::now()->getTimestamp());
```

Comment:
Is this timestamp used for anything mission-critical?  There's a bit of potential for timing-related errors here if it is - taking a timestamp in this method, passing it to another method and then (crucially) making a db call which could take an arbitrary amount of time to actually run might be problematic if the timestamp is meant to represent when the db operation runs (getting the db to generate the timestamp would solve that).

Not sure if this could ever be a problem as I don't know if this field is used for anything where timing accuracy is imperative but thought I'd ask.